src/transformer.ts

```diff
@@ -26,9 +26,37 @@ const buildTransformer = (parseMode: string) => {
       return prev(method, payload, signal);
     }
 
+    let newPayload = { ...payload, ...{ parse_mode: normalisedParseMode } };
+
+    if (
+      method === "editMessageMedia" ||
+      method === "answerInlineQuery"
+    ) {
+      if (
+        "media" in payload &&
+        !("parse_mode" in payload.media)
+      ) {
+        newPayload.media.parse_mode = normalisedParseMode;
+      }
+      else if ("results" in payload) {
+        for (let resulto in payload.results) {
+          let result = payload.results[resulto];
+          if (
+            "input_message_content" in result &&
+            !("parse_mode" in result.input_message_content)
+          ) {
+            newPayload.results[resulto].input_message_content.parse_mode = normalisedParseMode;
+          }
+          else if (!("parse_mode" in result)) {
+            newPayload.results[resulto].parse_mode = normalisedParseMode;
+          }
+        }
+      }
+    }
```

Comment:
There doesn't appear to be much similarities in logic between editMessageMedia and answerInlineQuery. We might benefit from separating how each case is handle with a switch statement instead.

```ts
switch(method) {
  case 'editMessageMedia':
    // ...
    break;
  case 'answerInlineQuery':
    // ...
    break;
  default:
    // ...
}
```

While changing it to the above form, we might want to consider excluding `"parse_mode" in payload` from the initial check and shifting that into the switch statement's default case instead.

Also consider [for...of](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...of) if you don't need the index of the array.

Reply:
I think index is required to update the payload, don't know if can use spreads, and avoid the loop.

Reply:
For...of is a loop syntax, not spread.

Reply:
If your concern is with the array order for the new object, there are things like `push`, taking advantage of the fact that the iteration is done in order.

Alternatively, `map` to operate on the entire input array to assign to the new object.

Reply:
Ok, is the updated changes correct ?